### PR TITLE
Purging: Broadcast purging and debit creation/squashing

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

**Comment by ericnewcomer** [comment]
Be weary of cascading delete with Msg.response_to

**Comment by rowanseymour** [comment]
Looked into that a bit more and `response_to` is only ever set to allow outgoing messages (thing being deleted) to reference incoming messages (not deleted) - so no problem

**Comment by ericnewcomer** [comment]
Ah, right, I guess that makes sense. Want to clone an AWS db and run this against it? Would also be interesting to know how long a full vacuum takes after it is complete so we can estimate our downtime when we go offline for this in the future.

**Comment by ericnewcomer** [comment]
Let's discuss the path forward for this on Monday.

File: temba/msgs/models.py [hunk 1]
```diff
@@ -143,6 +143,21 @@ class UnreachableException(Exception):
...
 
+class BroadcastRecipient(models.Model):
+    """
+    Through table for broadcast recipients many-to-many
+    """
+    broadcast = models.ForeignKey('msgs.Broadcast')
+
+    contact = models.ForeignKey(Contact)
+
+    purged_status = models.CharField(null=True, max_length=1,
```
Comment (ericnewcomer) [inline temba/msgs/models.py]: purged_status sounds to me like the status of purging, like whether it is was purged, etc. I think send_status or just status would be clearer.

Comment (rowanseymour) [inline temba/msgs/models.py]: What I was trying to get at here is that this field isn't used (it'll stay null) if the broadcast isn't purged, but am fine with with just status.

Comment (nicpottier) [inline temba/msgs/models.py]: Agree that status may be a little too ambiguous. `final_status`? I'm ok with `purged_status` as well.

Eric and I chatted and we think that storing all the statuses of purged messages is the right place to start. We can always squash again later if needbe and remove sent and wired, but let's keep it simple for this go.

Comment (rowanseymour) [inline temba/msgs/models.py]: If you're ok with `purged_status` then have left it as is, and have changed the purge task to record all message statuses

**Comment by ericnewcomer** [comment]
The migration conflict introduced by merging in master is addressed in  https://github.com/nyaruka/rapidpro/pull/986

**Comment by rowanseymour** [comment]
Added explanation to Sent and Failed views when org is enabled for purging:

screen shot 2016-12-16 at 10 34 35

**Pull request merged**